django/db/models/sql/query.py

```diff
@@ -1060,10 +1059,7 @@ def check_related_objects(self, field, value, opts):
             # opts would be Author's (from the author field) and value.model
             # would be Author.objects.all() queryset's .model (Author also).
             # The field is the related field on the lhs side.
-            # If _forced_pk isn't set, this isn't a queryset query or values()
-            # or values_list() was specified by the developer in which case
-            # that choice is trusted.
-            if (getattr(value, '_forced_pk', False) and
+            if (isinstance(value, Query) and not value.select and
```

Comment:
I guess we should be using `not (value.select or value.annotation_select_mask or value.extra_select_mask)` here as well.